package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./filter/IsEqual",
+  "./filter/IsIn",
+  "./filter/And",
+  "./filter/Or",
+  "./filter/Not"
+], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
+  "use strict";
+
+  /**
+   * @name RootFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @ignore
+   *
+   * @classdesc The `RootFilter` class implements a type of
+   * `AbstractTreeFilter` {@link pentaho.data.filter.AbstractTreeFilter}. Specifically
+   * the `RootFilter` extends the `RootFilter` {@link pentaho.data.filter.Or} filter.
+   *
+   * @description Creates a `RootFilter` filter that performs the union of a list of given `AbstractPropertyFilter`'s {@link pentaho.data.filter.AbstractPropertyFilter}.
+   *
+   * @param {Object} filterSpec The specification of a Filter.
+   *
+   */
+  var RootFilter = OrFilter.extend("pentaho.data.RootFilter", /** @lends pentaho.data.RootFilter# */{
+    constructor: function(filterSpec) {
+      if(filterSpec)
+        this.base(fromSpec(filterSpec));
+      else
+        this.base();
+    }
+  });
+
+  /**
+   * @name Filter
+   * @memberOf pentaho.data
+   * @class
+   * @amd pentaho/data/Filter
+   *
+   * @classdesc The `Filter` class provides factory methods for building the following filters:
+   *
+   * * {@link pentaho.data.filter.IsEqual}
+   * * {@link pentaho.data.filter.IsIn}
+   * * {@link pentaho.data.filter.Or}
+   * * {@link pentaho.data.filter.And}
+   * * {@link pentaho.data.filter.Not}
+   * * {@link pentaho.data.filter.RootFilter}
+   *
```

Comment:
It does not export RootFilter. In fact, RootFilter is just an artifice that we might kill off.
It exports a static `create` function that is a factory of RootFilters